addons/mail/static/src/core/message_service.js

```diff
@@ -144,6 +144,16 @@ export class MessageService {
         await this.orm.silent.call("mail.message", "set_message_done", [[message.id]]);
     }
 
+    async setPin(message, pinned) {
+        if (message.originThread.model !== "mail.channel") {
+            return;
+        }
```

Comment:
In which case do we allow a call on this method if the thread is not correct?

Reply:
Currently, never. This check is probably overkill. Removed.